src/primary_names.lua

```diff
@@ -237,29 +247,39 @@ end
 
 --- Get the address for a primary name, allowing for forward lookups (e.g. "foo.bar" -> "0x123")
 --- @param name string
---- @return WalletAddress|nil address - the address for the primary name, or nil if it does not exist
+--- @return WalletAddress|nil address -- the address for the primary name, or nil if it does not exist
 function primaryNames.getAddressForPrimaryName(name)
 	return PrimaryNames.names[name]
 end
 
 --- Get the name data for an address, allowing for reverse lookups (e.g. "0x123" -> "foo.bar")
 --- @param address string
---- @return PrimaryNameWithOwner|nil primaryNameWithOwner - the primary name with owner data, or nil if it does not exist
+--- @return PrimaryNameInfo|nil -- the primary name with owner data, or nil if it does not exist
 function primaryNames.getPrimaryNameDataWithOwnerFromAddress(address)
 	local nameData = PrimaryNames.owners[address]
 	if not nameData then
 		return nil
 	end
+
+	if not nameData.processId then
+		-- TODO: After initial deployment, we can make processId not nil and remove this block
+		local record = arns.getRecord(baseNameForName(nameData.name))
+		if record then
+			nameData.processId = record.processId
+			PrimaryNames.owners[address] = nameData
+		end
+	end
```

Comment:
nit - this isn't something that needs to be stored in state, it's cheap to compute on the fly and return with the name data